builtin/credential/aws/path_login.go

```diff
@@ -967,11 +984,16 @@ func (b *backend) pathLoginRenewIam(ctx context.Context, req *logical.Request, d
 					b.setCachedUserId(clientUserId, fullArn)
 				}
 			}
-			if !strutil.GlobbedStringsMatch(roleEntry.BoundIamPrincipalARN, fullArn) {
+			matchedWildcardBind := false
+			for _, principalARN := range roleEntry.BoundIamPrincipalARNs {
+				if strings.HasSuffix(principalARN, "*") && strutil.GlobbedStringsMatch(principalARN, fullArn) {
```

Comment:
Is prefix matching sufficient here, do you think, or should this support globs anywhere in the ARN?

Reply:
The prior code only supported prefix matching, so I'm just keeping that.

It's a good question as to whether globs anywhere in the ARN should be supported; I haven't seen any requests for it either on the mailing list or in the issue tracker, and if general wildcards should be supported, let's address that in a separate PR :)